src/KeyVault/KeyVault/Models/PSKeyVaultCertificatePolicy.cs

```diff
@@ -274,6 +281,147 @@ internal CertificatePolicy ToCertificatePolicy()
+            }else
+            {
+                certificatePolicy = new Track2Certificate.CertificatePolicy();
+            }
+            if (!string.IsNullOrWhiteSpace(SecretContentType))
+                certificatePolicy.ContentType = SecretContentType;
+
+            if (!string.IsNullOrWhiteSpace(Kty))
+                certificatePolicy.KeyType = Kty;
+
+            if (KeySize.HasValue)
+                certificatePolicy.KeySize = KeySize;
+
+            if (!string.IsNullOrWhiteSpace(Curve))
+                certificatePolicy.KeyCurveName = Curve;
+
+            if (ReuseKeyOnRenewal.HasValue)
+                certificatePolicy.ReuseKey = ReuseKeyOnRenewal;
+
+            if (Exportable.HasValue)
+                certificatePolicy.Exportable = Exportable;
+
+            if (CertificateTransparency.HasValue)
+                certificatePolicy.CertificateTransparency = CertificateTransparency;
+
+            if (!string.IsNullOrWhiteSpace(CertificateType))
+                certificatePolicy.CertificateType = CertificateType;
+
+            if (Enabled.HasValue)
+                certificatePolicy.Enabled = Enabled;
+
+            if (ValidityInMonths.HasValue)
+                if (ValidityInMonths.HasValue)
+                    certificatePolicy.ValidityInMonths = ValidityInMonths.Value;
+
+            if (RenewAtNumberOfDaysBeforeExpiry.HasValue ||
+                RenewAtPercentageLifetime.HasValue ||
+                EmailAtNumberOfDaysBeforeExpiry.HasValue ||
+                EmailAtPercentageLifetime.HasValue)
+            {
+                if ((RenewAtNumberOfDaysBeforeExpiry.HasValue ? 1 : 0) +
+                    (RenewAtPercentageLifetime.HasValue ? 1 : 0) +
+                    (EmailAtNumberOfDaysBeforeExpiry.HasValue ? 1 : 0) +
+                    (EmailAtPercentageLifetime.HasValue ? 1 : 0) > 1)
+                {
+                    throw new ArgumentException("Only one of the values for RenewAtNumberOfDaysBeforeExpiry, RenewAtPercentageLifetime, EmailAtNumberOfDaysBeforeExpiry, EmailAtPercentageLifetime  can be set.");
+                }
+
+                if (RenewAtNumberOfDaysBeforeExpiry.HasValue)
+                {
+                    certificatePolicy.LifetimeActions.Add(
+                        new Track2Certificate.LifetimeAction(new Track2Certificate.CertificatePolicyAction("AutoRenew"))
+                        {
+                            DaysBeforeExpiry = RenewAtNumberOfDaysBeforeExpiry
+                        }
+                    );
+                }
+
+                if (RenewAtPercentageLifetime.HasValue)
+                {
+                    certificatePolicy.LifetimeActions.Add(
+                        new Track2Certificate.LifetimeAction(new Track2Certificate.CertificatePolicyAction("AutoRenew"))
+                        {
+                            LifetimePercentage = RenewAtPercentageLifetime
```

Comment:
```suggestion
                            LifetimePercentage = RenewAtPercentageLifetime.Value
```
Please check all properties about .Value

Reply:
As explained before, all these situations have exactly the same data type at either side of the equal sign. Maybe the change is unnecessary. Please let me know if you still think the changes are necessary. Thank you.